sql/6_0_0-to-6_1_0_upgrade.sql

```diff
@@ -248,3 +248,45 @@ ALTER TABLE `procedure_providers` ADD `type` VARCHAR(31) DEFAULT NULL;
 #IfMissingColumn procedure_answers procedure_code
 ALTER TABLE `procedure_answers` ADD `procedure_code` VARCHAR(31) DEFAULT NULL;
 #EndIf
+
+#IfNotRow users username oe-system
+INSERT INTO `users`(`username`,`password`,`lname`,`authorized`,`active`) VALUES ('oe-system','NoLogin','System Operation User',0,0);
+INSERT INTO `gacl_aro`(`id`, `section_value`, `value`, `order_value`, `name`, `hidden`)
+    SELECT max(`id`)+1,'users','oe-system',10,'System Operation User', 0 FROM `gacl_aro`;
+INSERT INTO `gacl_groups_aro_map`(`group_id`, `aro_id`)
+    VALUES (
+        (SELECT `id` FROM `gacl_aro_groups` WHERE parent_id=10 AND value='admin')
+        ,(SELECT `id` FROM `gacl_aro` WHERE `section_value` = 'users' AND `value` = 'oe-system')
+    );
+#EndIf
+
+#IfNotTable
+CREATE TABLE `export_job` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `uuid` binary(16) DEFAULT NULL ,
+  `user_id` varchar(40) NOT NULL,
+  `client_id` varchar(80) NOT NULL,
+  `status` varchar(40) NOT NULL,
+  `start_time` datetime DEFAULT NULL,
+  `resource_include_time` datetime DEFAULT NULL,
+  `output_format` varchar(128) NOT NULL,
+  `request_uri` varchar(128) NOT NULL,
+  `resources` text,
+  `output` text,
+  `errors` text,
+  `access_token_id` text,
+  PRIMARY KEY  (`id`),
+  UNIQUE (`uuid`)
+) ENGINE=InnoDB COMMENT='fhir export jobs';
+#EndIf
+
+#IfNotRow categories name FHIR Export Document
+SET @max_rght = (SELECT MAX(rght) FROM categories);
+INSERT INTO categories(`id`,`name`, `value`, `parent`, `lft`, `rght`, `aco_spec`) select (select MAX(id) from categories) + 1, 'FHIR Export Document', '', 1, @max_rght, @max_rght + 1, 'admin|super' from categories where name = 'Categories';
```

Comment:
i think we can now simplify this line to:
```
INSERT INTO categories (`id`,`name`, `value`, `parent`, `lft`, `rght`, `aco_spec`) VALUES ((select MAX(id) from categories) + 1, 'FHIR Export Document', '', 1, @max_rght, @max_rght + 1, 'admin|super');
```

Reply:
Nope, you get this error: 
```
INSERT INTO categories (`id`,`name`, `value`, `parent`, `lft`, `rght`, `aco_spec`) VALUES ((select MAX(id) from categories) + 1, 'FHIR Export Document', '', 1, @max_rght, @max_rght + 1, 'admin|super')<br />
SQL Error with statement:query failed: INSERT INTO categories (`id`,`name`, `value`, `parent`, `lft`, `rght`, `aco_spec`) VALUES ((select MAX(id) from categories) + 1, &#039;FHIR Export Document&#039;, &#039;&#039;, 1, @max_rght, @max_rght + 1, &#039;admin|super&#039;)--Table &#039;categories&#039; is specified twice, both as a target for &#039;INSERT&#039; and as a separate source for data==&gt;/var/www/localhost/htdocs/openemr/library/sql_upgrade_fx.php at 1175:sqlStatement
```

So I'll leave it be.

Reply:
oops, my bad on that.